src/PointEvalHandler.jl

```diff
@@ -45,20 +40,27 @@ function Base.show(io::IO, ::MIME"text/plain", ph::PointEvalHandler)
     end
 end
 
-function PointEvalHandler(grid::AbstractGrid, points::AbstractVector{Vec{dim,T}}; search_nneighbors=3, warn=true) where {dim, T}
+# Internals:
+# `PointEvalHandler` takes the following keyword arguments:
+#  - `search_nneighbors`: How many nodes should be found in the nearest neighbor search for each
+#    point. Usually there is no need to change this setting. Default value: `3`.
+#  - `warn::Bool`: Show a warning if a point is not found. Default value: `true`.
+#  - `newton_max_iters::Int`: Maximum number of inner Newton iterations. Default value: `10`.
+#  - `newton_residual_tolerance`: Tolerance for the residual norm to indicate convergence in the
+#    inner Newton solver. Default value: `1e-10`.
+function PointEvalHandler(grid::AbstractGrid{dim}, points::AbstractVector{Vec{dim,T}}; search_nneighbors=3, warn::Bool=true, newton_max_iters::Int = 10, newton_residual_tolerance::T = 1e-10) where {dim, T}
     node_cell_dicts = _get_node_cell_map(grid)
-    cells, local_coords = _get_cellcoords(points, grid, node_cell_dicts, search_nneighbors, warn)
+    cells, local_coords = _get_cellcoords(points, grid, node_cell_dicts, search_nneighbors, warn, #=linesearch_max_substeps=# 5, newton_max_iters, newton_residual_tolerance)
     return PointEvalHandler(grid, cells, local_coords)
 end
 
-function _get_cellcoords(points::AbstractVector{Vec{dim,T}}, grid::AbstractGrid, node_cell_dicts::Dict{C,Dict{Int, Vector{Int}}}, search_nneighbors, warn) where {dim, T<:Real, C}
-
+function _get_cellcoords(points::AbstractVector{Vec{dim,T}}, grid::AbstractGrid, node_cell_dicts::Dict{C,Dict{Int, Vector{Int}}}, search_nneighbors, warn, linesearch_max_substeps, max_iters, tol_norm) where {dim, T<:Real, C}
```

Comment:
keyword arguments?

Reply:
Why?

Reply:
Seems easier to read when we have so many, instead of putting the `#= name =#` comment, and they are first kwarg, then arg, and then kwarg again later. 

We are also now defining the defaults differently for `PointEvalHandler` compared to `find_local_coordinate` (at least `linesearch_max_substeps` and `warn`), is this intended?

Perhaps it would be nice to define the defaults only in one location?

Reply:
But if you disagree, can be discussed and potentially done separately! 
Don't let this stop the PR - it is just an internal detail after all, and I guess you intend the keywords to PointEvalHandler also as internal for now based on the comment?

Reply:
I added a struct to gather these. Might also be interesting as an option to swap the solver.

Reply:
Indeed, the kwargs are internals for now. We can make them later public once the interfaces settle